web/public/js/directives/cd-rounded-image/index.js

```diff
@@ -40,16 +39,16 @@ function cdRoundedImage(){
           if (newId) {
             cdRI.srcId = newId;
             cdRI.src += $scope.srcId ? $scope.srcId : '';
-            idWatcher();
-            if(_.isUndefined(cdRI.format)) {
-              //  TODO: use a provider like cd-Entity-Service (i.e: cd-dojos-service=>getAvatar) to get the proper image
-              $http.head(cdRI.src).then(function successCallback(response) {
-                //Nothing to do, it all works as expected, the image is stored online
-              }, function errorCallback(response) {
-                //File doesn't exists, we should fallback
+            $timeout(function () {
+              var image = angular.element(
+                document.getElementsByClassName(cdRI.edit ?
```

Comment:
Potential to trigger on error of a different image if directive is used multiple times on one page.

Inject `$element` and use `$element.find` instead of `document.getElementsByClassName`. Limits the search to this directive's template only.